### DISCUSSION TITLE
how to restore transport order when restart the kernel

### DISCUSSION THREAD

**Discussion by weibin06** [openTCS/opentcs#74] [Q&A]
How to restore an order after restarting the opentcs kernel. This is very important, once the order is lost, the vehicle may be out of control.

**Comment by swltr** [comment]
openTCS does not do that by itself. There are two options:

* After restarting the kernel, the system that created the original transport order creates it again.
* A custom kernel extension persists transport order data and materializes the transport orders on restart if necessary.
